examples/example_server_forum.py

```diff
@@ -0,0 +1,155 @@
+import socket
+import sys
+import socketserver
+from http.server import BaseHTTPRequestHandler
+from http.server import HTTPServer
+import urllib.parse
+
+# using port numbers prepended with 9 to avoid calling sudo during test
+servport = 9090
+posts_file = "./forum/posts.txt"
+
+def jsonify(user, msg):
+    max_chars = 256
+    num_attributes = 2
+    max_chars -= (num_attributes*6+1) # "":"", and {} but last line has no comma
+    return "{\"user\":\"" + user + "\",\"msg\":\"" + msg + "\"}";
+
+# I did this in case any messages have newline characters (so I won't necessarily store comments as one line the posts file)
+def read_simple_json(json_str, start_idx):
+    return_dict = {}
+    in_quotes = False
+    in_esc = False
+    quote_start = None
+    param = None
+    seeker = start_idx
+    length = len(json_str)
+    if start_idx >= length or json_str[start_idx] != "{":
+        return None
+    while seeker < length:
+        cur_char = json_str[seeker]
+        if not in_quotes and not in_esc and cur_char == "}":
+            break
+        elif in_esc:
+            in_esc = False
+        elif cur_char == "\\":
+            in_esc = True
+        # handle quotes
+        elif cur_char == "\"":
+            if not in_quotes:
+                quote_start = seeker
+            if in_quotes:
+                if param == None and seeker+1 < length and json_str[seeker+1] == ":":
+                    param = json_str[quote_start+1:seeker]
+                else:
+                    return_dict[param] = json_str[quote_start+1:seeker]
+                    param = None
+                quote_start = None
+            in_quotes = not in_quotes
+        
+        seeker += 1
+    # end of while loop
+    return_dict['_length'] = seeker+1 - start_idx # how many chars to seek past in order to go past this json object
+    return return_dict
+
+# https://wiki.python.org/moin/BaseHttpServer
+class MyHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        print("")
+        print("GET for", self.client_address)
+        print("Path:", self.path)
+
+        output_txt = ""
+        data_dict = {}
+        try:
```

Comment:
I don't think you need to put this in a try block since there isn't a following except statement

Reply:
The "try" block for here is in case there is no question mark (no params in the GET url), index will throw an error. It does run when I send such a request, so I am not sure if I need an except.